### PR TITLE
Fix gufunc kwargs support

### LINKED ISSUE #6773: Regression in 0.53rc2: axis= parameter in gufuncs
This broke in 0.53rc2; it was working in 0.52:

```python
import numba
import numpy

@numba.guvectorize(["f8[:],f8[:]"], "(n)->(n)")
def my_cumsum(x, res):
    acc = 0
    for i in range(x.shape[0]):
        acc += x[i]
        res[i] = acc


x = numpy.ones((20, 30))
y = my_cumsum(x, axis=0)
```
Output:
```
TypeError: __call__() got an unexpected keyword argument 'axis'
```

### PR DESCRIPTION
**Pull request by sklam**
Fix #6773

Fixes kwargs support for gufunc.

### CODE REVIEW

**Comment by esc** [comment]
Buildfarm ID: `numba_smoketest_cpu_yaml_9`

File: numba/tests/npyufunc/test_gufunc.py [hunk 1]
```diff
@@ -62,6 +62,24 @@ def test_ufunc_like(self):
 
         np.testing.assert_equal(out, x * x + x)
 
+    def test_axis(self):
+        @guvectorize(["f8[:],f8[:]"], "(n)->(n)")
```
Comment (esc) [inline numba/tests/npyufunc/test_gufunc.py:66]: Maybe include a comment here pointing to: https://github.com/numba/numba/issues/6773 -- IIRC this is done quite often for tests that expose specific bugs in the Numba code-base.

File: numba/np/ufunc/gufunc.py [hunk 2]
```diff
@@ -133,8 +136,7 @@ def __call__(self, *args):
         # at this point we know the gufunc is a dynamic one
         ewise = self._get_ewise_dtypes(args)
         if not (self.ufunc and ufunc_find_matching_loop(self.ufunc, ewise)):
-            self._is_dynamic = True
```
Comment (esc) [inline numba/np/ufunc/gufunc.py:136]: This is odd. Is this fixing an oversight unrelated to kwarg processing?

File: numba/tests/npyufunc/test_gufunc.py [hunk 3]
```diff
@@ -140,6 +162,24 @@ def sum_row(inp, out):
         with self.assertRaisesRegex(TypeError, msg):
             sum_row(inp)
 
+    def test_axis(self):
+        @guvectorize("(n)->(n)")
+        def my_cumsum(x, res):
+            acc = 0
```
Comment (esc) [inline numba/tests/npyufunc/test_gufunc.py:168]: Same comment as above.

**Review by esc — CHANGES_REQUESTED** [review]
One question, one suggestion.

**Comment by esc** [comment]
So, this fails on the farm with: `[2021-03-03 08:19:54,958] {docker_operator.py:262} INFO - conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-ppc64le: {"llvmlite[version='>=0.37.0dev0,<0.37']"}` -- I presume this is because there is no correct `llvmlite` on the `numba/label/main` channel, but only on the `numba/label/dev`. I'll have to check.

**Comment by esc** [comment]
My assumption turns out to be correct. `conda build -c numba/label/main --no-test --python=3.7 --numpy=1.14 buildscripts/condarecipe.local` is the build command and it doesn't include `numba/label/main` as channel.

**Comment by esc** [comment]
OK, after adding `numba/label/dev` to the `conda build` command, I still get:

`[2021-03-03 03:12:26,372] {bash_operator.py:137} INFO - conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform win-32: {"llvmlite[version='>=0.37.0dev0,<0.37']"}`

**Comment by esc** [comment]
it's strange, because the 0.37.0dev0 does seem to be available on anaconda.org:

https://anaconda.org/numba/llvmlite/files?version=0.37.0dev0

**Comment by esc** [comment]
OK, it may be fixed, linux, windows and OSX have their own config for the channels to use during build.

**Comment by esc** [comment]
Happy to report this passed on the Anaconda internal build farm.

File: numba/np/ufunc/gufunc.py (hunk 2 above)
Comment (sklam) [reply numba/np/ufunc/gufunc.py:136]: Correct

File: numba/tests/npyufunc/test_gufunc.py (hunk 1 above)
Comment (sklam) [reply numba/tests/npyufunc/test_gufunc.py:66]: will do

Comment (sklam) [reply numba/tests/npyufunc/test_gufunc.py:67]: done in https://github.com/numba/numba/pull/6786/commits/20c272d5d387c477197cd3bee18abc6adc445a32

File: numba/tests/npyufunc/test_gufunc.py (hunk 3 above)
Comment (sklam) [reply numba/tests/npyufunc/test_gufunc.py:170]: done in https://github.com/numba/numba/pull/6786/commits/20c272d5d387c477197cd3bee18abc6adc445a32

**Review by esc — APPROVED** [review]

File: numba/tests/npyufunc/test_gufunc.py [hunk 4]
```diff
@@ -140,6 +163,25 @@ def sum_row(inp, out):
         with self.assertRaisesRegex(TypeError, msg):
             sum_row(inp)
 
+    def test_axis(self):
+        # issue https://github.com/numba/numba/issues/6773
+        @guvectorize("(n)->(n)")
```
Comment (stuartarchibald) [inline numba/tests/npyufunc/test_gufunc.py:168]: This isn't the exact reproducer from the ticket as there's no type signature provided?

Comment (sklam) [reply numba/tests/npyufunc/test_gufunc.py:168]: This is the dynamic gufunc version of it. The exact one is at https://github.com/numba/numba/pull/6786/files#diff-53626a767921908cedfa774172dffeebb669f76bb7ee26af3bc51c37987d5bfdR67, but both version are doing through the same `GUFunc.__call__`. So this is the variant to make sure that the dynamic gufunc is handling the kwargs correctly as well.

Comment (stuartarchibald) [reply numba/tests/npyufunc/test_gufunc.py:168]: ah thanks, missed that.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch, looks good.

**Pull request merged**